sequential/backend/node_mdd.hpp

```diff
@@ -29,7 +29,16 @@ class Node {
 	~Node() {
 		for (int i = 0; i < children.size(); i++){
 			delete children[i];
-		} 
+		}
+
+		for (int i=0; i < attr.size(); i++){
+		    for (int j = 0; j < attr[i]->size(); j++)
+					delete attr[i]->at(j);
+		    delete attr[i];
+		}
+		std::vector<vector<Node*>*>().swap(children);
```

Comment:
can you add a commentary line on these swap operations? Since it is net-new, it would be good to capture what is the idea behind

Reply:
Yes, I can add a comment to this part. This is one of the ways to free the memory taken by the vector object.  The vector is swapped with an empty vector that has no allocated memory, then this can better deallocate the memory. [Here](https://www.techiedelight.com/delete-vector-free-memory-cpp/) I find some comparison between different ways to deallocate the memory of a vector.

Reply:
interesting.. thanks for sharing,  Iwas not aware of the shrink_to_fit() method